### ISSUE TITLE
Resolver class fails to register with provider wrapper that specifies useFactory

### ISSUE THREAD

**Issue by carlevans719** [nestjs/graphql#1700]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

When including a class decorated with `@Resolver()` via a FactoryProvider, it is not registered with the Nest.js GraphQL module

## Expected behavior

Expected that useFactory would work the same way as passing the resolver class directly into the providers array. Selected documentation issue / request as unsure if this approach is supported or not. [docs for reference](https://docs.nestjs.com/graphql/resolvers#module)

## Minimal reproduction of the problem with instructions

[Minimal reproduction - steps in README](https://github.com/carlevans719/nest-gql-use-factory-repro)

## Environment

<pre><code>
Nest version: 8.0.6
Tested in 7.x.x too - same behaviour
 
For Tooling issues:
- Node version: 12.16.2
- Platform:  Linux

Others:
    "@apollo/gateway": "^0.38.0",
    "@nestjs/common": "^8.0.6",
    "@nestjs/core": "^8.0.6",
    "@nestjs/graphql": "^8.0.2",
    "graphql": "^15.5.1",
    "reflect-metadata": "^0.1.13",
</code></pre>

**Comment by jmcdo29** [comment]
What's the use case for trying to register a resolver via a factory?

**Comment by carlevans719** [comment]
I have a config service that specifies whether certain GQL queries are enabled/disabled. I've got those queries split out into a separate resolver that I'm trying to conditionally register with nest depending on the value from the config service.

My thinking is that I can use the `inject` prop with `useFactory` to either return the resolver, or null / a dummy resolver based on the value in the config service.

A couple of observations that are unrelated to my use case (I think):

- When `autoSchemaFile === true`, **no** provider wrapper works (useClass, useValue, useFactory).
- When `autoSchemaFile !== true`, only useClass works

I've added comments to highlight this in the repro repo.

Let me know if it'd help to add a slightly more fleshed-out example of my use case on a branch in the repro
